elements/lisk-cryptography/test/ed.spec.ts

```diff
@@ -30,219 +28,165 @@ import { MAX_UINT32 } from '../src/constants';
 
 const changeLength = (buffer: Buffer): Buffer => Buffer.concat([Buffer.from('00', 'hex'), buffer]);
 
-describe('sign', () => {
-	const tag = createMessageTag('TST');
-	const chainID = Buffer.from('10000000', 'hex');
-	const defaultPrivateKey =
-		'314852d7afb0d4c283692fef8a2cb40e30c7a5df2ed79994178c10ac168d6d977ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588';
-	const defaultPublicKey = '7ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588';
-	const defaultMessage = 'Some default text.';
-	const defaultSignature =
-		'68937004b6720d7e1902ef05a577e6d9f9ab2756286b1f2ae918f8a0e5153c15e4f410916076f750b708f8979be2430e4cfc7ebb523ae1905d2ea1f5d24ce700';
-	const defaultPrintedMessage = `
+const privateKey = Buffer.from(
+	'314852d7afb0d4c283692fef8a2cb40e30c7a5df2ed79994178c10ac168d6d977ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588',
+	'hex',
+);
+const publicKey = Buffer.from(
+	'7ef45cd525e95b7a86244bbd4eb4550914ad06301013958f4dd64d32ef7bc588',
+	'hex',
+);
+
+describe('sign and verify message', () => {
+	const message = 'Some default text.';
+	const { signature } = signMessageWithPrivateKey(message, privateKey);
+	const printedMessage = `
 -----BEGIN LISK SIGNED MESSAGE-----
 -----MESSAGE-----
-${defaultMessage}
+${message}
 -----PUBLIC KEY-----
-${defaultPublicKey}
+${publicKey.toString('hex')}
 -----SIGNATURE-----
-${defaultSignature}
+${signature.toString('hex')}
 -----END LISK SIGNED MESSAGE-----
 `.trim();
 
-	const defaultData = Buffer.from('This is some data');
-	const defaultDataSignature =
-		'2c7b4e67bf6a7030de9a7454e5e16b8d23dcee1f5ea4b8b881c89ec5834d534afe3d791ab2e45df331d64493537ce57448025d09cfdba39ba7edeed505efb602';
-
-	let defaultSignedMessage: SignedMessageWithPrivateKey;
-
-	beforeEach(() => {
-		defaultSignedMessage = {
-			message: defaultMessage,
-			publicKey: Buffer.from(defaultPublicKey, 'hex'),
-			signature: Buffer.from(defaultSignature, 'hex'),
-		};
-	});
-
 	describe('#signMessageWithPrivateKey', () => {
-		it('should create a signed message using a secret private key', () => {
-			const signedMessage = signMessageWithPrivateKey(
-				defaultMessage,
-				Buffer.from(defaultPrivateKey, 'hex'),
-			);
-			expect(signedMessage).toEqual(defaultSignedMessage);
+		it('should create a signed message using a private key', () => {
+			const signedMessage = signMessageWithPrivateKey(message, privateKey);
+			expect(signedMessage).toMatchSnapshot();
```

Comment:
I think we should ass test where we specify different tag

Reply:
Added unit tests both for custom tags, and for messages provided in Buffer format.